src/io/files.js

```diff
@@ -820,55 +794,103 @@ p5.prototype.httpGet = function () {
  */
 p5.prototype.httpPost = function () {
   var args = new Array(arguments.length);
-  for (var i = 0; i < args.length; ++i) {
+  args[0] = arguments[0];
+  args[1] = 'POST';
+  for (var i = 2; i < args.length; ++i) {
     args[i] = arguments[i];
   }
-  args.push('POST');
   p5.prototype.httpDo.apply(this, args);
 };
 
 /**
  * Method for executing an HTTP request. If data type is not specified,
  * p5 will try to guess based on the URL, defaulting to text.<br><br>
- * You may also pass a single object specifying all parameters for the
- * request following the examples inside the reqwest() calls here:
- * <a href='https://github.com/ded/reqwest#api'>
- * https://github.com/ded/reqwest#api</a>
+ * For more advanced use, you may also pass in the path as the first argument
+ * and a object as the second argument, the signature follows the one specified
+ * in the Fetch API specification.
  *
  * @method httpDo
  * @param  {String}        path       name of the file or url to load
  * @param  {String}        [method]   either "GET", "POST", or "PUT",
  *                                    defaults to "GET"
- * @param  {Object}        [data]     param data passed sent with request
  * @param  {String}        [datatype] "json", "jsonp", "xml", or "text"
+ * @param  {Object}        [data]     param data passed sent with request
  * @param  {Function}      [callback] function to be executed after
  *                                    httpGet() completes, data is passed in
  *                                    as first argument
  * @param  {Function}      [errorCallback] function to be executed if
  *                                    there is an error, response is passed
  *                                    in as first argument
  */
+
+/**
+ * @method httpDo
+ * @param  {String}        path
+ * @param  {Object}        options   Request object options as documented in the
+ *                                    "fetch" API
+ * <a href="https://developer.mozilla.org/en/docs/Web/API/Fetch_API">reference</a>
+ * @param  {Function}      [callback]
+ * @param  {Function}      [errorCallback]
+ */
```

Comment:
Yeah, sure. I think to highlight what would be a good circumstance to use the request object (needing to set headers), maybe have it request a server that uses some headers. Any API you can think of might be a good example?